### PR TITLE
♻️Refactor Navbar and Statusbar

### PR DESCRIPTION
**Pull request by Vyperus**
## What did you change?

This branch adds a NavBar component that is used in all views.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: npm-debug.log [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+23 error If you do, this is most likely a problem with the bpmn-studio package,
+23 error not with npm itself.
+23 error Tell the author that this fails on your system:
+23 error     ./bin/bpmn-studio.js
+23 error You can get information on how to open an issue for this project with:
+23 error     npm bugs bpmn-studio
+23 error Or if that isn't available, you can get their info via:
+23 error     npm owner ls bpmn-studio
+23 error There is likely additional logging output above.
+24 verbose exit [ 1, true ]
```
Comment (cmg-dev) [inline npm-debug.log]: hä?

File: src/app.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ export class App {
 
   public router: Router;
   public environment: any = environment;
+  public showSolutionExplorer: boolean = false;
```
Comment (cmg-dev) [inline src/app.ts]: 👍

File: src/app.html [hunk 3]
```diff
@@ -3,6 +3,17 @@
...
   <require from="./app.css"></require>
   <require from="toastr/build/toastr.min.css"></require>
+  <require from="./modules/navbar/navbar"></require>
+  <require from="./modules/process-solution-panel/process-solution-panel"></require>
+
+  <nav-bar show-solution-explorer.two-way="showSolutionExplorer" ref="navbar"></nav-bar>
+  <div class="bpmn-studio-layout">
+    <div>
+      <process-solution-panel show.bind="showSolutionExplorer"></process-solution-panel>
+    </div>
```
Comment (cmg-dev) [inline src/app.html]: sind die divs sinnvoll?

File: src/app.html [hunk 4]
```diff
@@ -3,6 +3,17 @@
...
+    <div class="router-view">
+      <router-view></router-view>
+    </div>
```
Comment (cmg-dev) [inline src/app.html]: die divs sollten auch überflüssig sein.

File: src/app.html [hunk 5]
```diff
@@ -3,6 +3,17 @@
   <require from="font-awesome/css/font-awesome.min.css"></require>
   <require from="./app.css"></require>
   <require from="toastr/build/toastr.min.css"></require>
+  <require from="./modules/navbar/navbar"></require>
+  <require from="./modules/process-solution-panel/process-solution-panel"></require>
+
+  <nav-bar show-solution-explorer.two-way="showSolutionExplorer" ref="navbar"></nav-bar>
```
Comment (cmg-dev) [inline src/app.html]: 👍 

noch eine leere Zeile hier drunter und dann <3

File: src/app.html [hunk 6]
```diff
@@ -3,6 +3,17 @@
   <require from="font-awesome/css/font-awesome.min.css"></require>
   <require from="./app.css"></require>
   <require from="toastr/build/toastr.min.css"></require>
+  <require from="./modules/navbar/navbar"></require>
+  <require from="./modules/process-solution-panel/process-solution-panel"></require>
```
Comment (cmg-dev) [inline src/app.html]: Sortierung!

1. Framework Kram
2. 3rd Party Kram
3. Lokaler Kram

File: npm-debug.log (hunk 1 above)
Comment (cmg-dev) [reply npm-debug.log]: viel spaß beim löschen :P

File: src/modules/processdef-list/processdef-list.html [hunk 7]
```diff
@@ -4,52 +4,6 @@
...
-      <div class="menu-bar__menu menu-bar__menu--center"></div>
-
-      <div class="menu-bar__menu menu-bar__menu--right">
-        <user-login></user-login>
-      </div>
-    </div>
-
-    <process-solution-panel show.bind="solutionExplorerIsShown"></process-solution-panel>
-
     <div class="processdef-list-container">
```
Comment (cmg-dev) [inline src/modules/processdef-list/processdef-list.html]: du musst etwas vorsichtig sein, das wird einen großen konflikt mit dem branch von steffen und mir erzeugen!

Comment (cmg-dev) [reply src/modules/processdef-list/processdef-list.html]: lass das morgen zusammen mergen und den hier rebasen.

File: src/modules/navbar/navbar.ts [hunk 8]
```diff
@@ -0,0 +1,87 @@
+import {EventAggregator} from 'aurelia-event-aggregator';
+import {bindable, inject} from 'aurelia-framework';
+import {RouteConfig, Router} from 'aurelia-router';
+import { IProcessDefEntity } from '../../contracts';
```
Comment (alexanderkasten) [inline src/modules/navbar/navbar.ts]: Remove whitespaces

File: src/modules/navbar/navbar.ts [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+import environment from '../../environment';
+
+@inject(Router, EventAggregator)
+export class NavBar {
+  private router: Router;
```
Comment (alexanderkasten) [inline src/modules/navbar/navbar.ts]: Prefix private variables with an underscore, also methods.

File: src/modules/process-list/process-list.html [hunk 10]
```diff
@@ -2,58 +2,6 @@
   <require from="./process-list.css"></require>
   <require from="../process-solution-panel/process-solution-panel"></require>
```
Comment (alexanderkasten) [inline src/modules/process-list/process-list.html]: Is it used here ?

File: src/modules/processdef-detail/processdef-detail.html [hunk 11]
```diff
@@ -6,124 +6,7 @@
   <require from="../user-login/user-login"></require>
   <require from="../process-solution-panel/process-solution-panel"></require>
```
Comment (alexanderkasten) [inline src/modules/processdef-detail/processdef-detail.html]: same as above, is it used ? if not remove it please

File: src/modules/processdef-start/processdef-start.html [hunk 12]
```diff
@@ -3,58 +3,6 @@
   <require from="../dynamic-ui-wrapper/dynamic-ui-wrapper"></require>
   <require from="../process-solution-panel/process-solution-panel"></require>
```
Comment (alexanderkasten) [inline src/modules/processdef-start/processdef-start.html]: same as above

File: src/modules/status-bar/status-bar.ts [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+import environment from '../../environment';
+
+@inject(EventAggregator)
+export class StatusBar {
+
+  public baseRoute: string = environment.bpmnStudioClient.baseRoute;
+  public showXMLButton: boolean = false;
+  public xmlIsShown: boolean = false;
+
+  private eventAggregator: EventAggregator;
```
Comment (alexanderkasten) [inline src/modules/status-bar/status-bar.ts]: prefix this with an underscore

File: src/modules/task-list/task-list.html [hunk 14]
```diff
@@ -4,58 +4,6 @@
   <require from="../process-solution-panel/process-solution-panel"></require>
```
Comment (alexanderkasten) [inline src/modules/task-list/task-list.html]: same as above

**Pull request merged**